packages/flutter/lib/src/cupertino/localizations.dart

```diff
@@ -0,0 +1,221 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+
+/// Defines the localized resource values used by the Cupertino widgets.
+///
+/// See also:
+///
+///  * [DefaultCupertinoLocalizations], the default, English-only, implementation
+///    of this interface.
+// TODO(xster): Supply non-english strings.
+abstract class CupertinoLocalizations {
+  /// The given number in local writing system.
+  ///
+  /// Different languages can have different numeral systems, and this function
+  /// will return the given number written in the local language.
+  /// Example of some numeral systems can be found here:
+  /// https://en.wikipedia.org/wiki/List_of_numeral_systems
+  String number(int num);
+
+  /// Month that is shown in [CupertinoDatePicker] corresponding to the given
+  /// month index.
+  ///
+  /// Examples: datePickerMonth(1) in:
+  ///
+  ///  - US English: January
+  ///  - Korean: 1월
+  String datePickerMonth(int monthIndex);
+
+  /// Day of month that is shown in [CupertinoDatePicker] corresponding to
+  /// the given day index.
+  ///
+  ///
+  /// Examples: datePickerDayOfMonth(1) in:
+  ///
+  ///  - US English: 1
+  ///  - Korean: 1일
+  String datePickerDayOfMonth(int dayIndex);
+
+  /// Year that is shown in [CupertinoDatePicker] corresponding to the given
+  /// year index.
+  ///
+  ///
+  /// Examples: datePickerDayOfMonth(1) in:
+  ///
+  ///  - US English: 2018
+  ///  - Korean: 2018년
+  String datePickerYear(int yearIndex);
+
+  /// Formats the date using a medium-width format.
+  ///
+  /// Abbreviates month and days of week. This appears in the date spinner of
+  /// [CupertinoDatePicker].
+  ///
+  /// Examples:
+  ///
+  /// - US English: Wed Sep 27
+  /// - Russian: ср сент. 27
+  String datePickerMediumDate(DateTime date);
+
+  /// The order of the date elements that will be shown in [CupertinoDatePicker].
+  /// Can be any permutation of 'DMY' ('D': day, 'M': month, 'Y': year).
+  String get datePickerDateOrder;
+
+  /// The abbreviation for ante meridiem (before noon) shown in the time picker.
+  String get anteMeridiemAbbreviation;
+
+  /// The abbreviation for post meridiem (after noon) shown in the time picker.
+  String get postMeridiemAbbreviation;
+
+  /// Label that appears next to the hour picker in
+  /// [CupertinoCountdownTimerPicker] when selected hour value is ```hour```.
+  /// This function will deal with pluralization based on the ```hour``` parameter.
+  String timerPickerHourLabel(int hour);
```

Comment:
I'm not sure if this is the best way to deal with pluralization. In English and many other common languages, there are at most 2 forms: singular and plural. However, that might not be true for all languages, so here I leave it as general as it can be, and let the overriding class deal with it.
What do you think @xster?

Reply:
See the selectedRowCountTitle example
https://github.com/flutter/flutter/blob/master/packages/flutter_localizations/lib/src/l10n/README.md

Reply:
yup, the example has the same pattern.

Reply:
Not sure if I answered the question. Having the a simpler DefaultCupertinoLocalization for english is fine. What does iOS do when it's 0?